# Housing Rental Analysis for San Francisco

In this challenge, your job is to use your data visualization skills, including aggregation, interactive visualizations, and geospatial analysis, to find properties in the San Francisco market that are viable investment opportunities.

## Instructions

Use the `san_francisco_housing.ipynb` notebook to visualize and analyze the real-estate data.

Note that this assignment requires you to create a visualization by using hvPlot and GeoViews. Additionally, you need to read the `sfo_neighborhoods_census_data.csv` file from the `Resources` folder into the notebook and create the DataFrame that you’ll use in the analysis.

The main task in this Challenge is to visualize and analyze the real-estate data in your Jupyter notebook. Use the `san_francisco_housing.ipynb` notebook to complete the following tasks:

* Calculate and plot the housing units per year.

* Calculate and plot the average prices per square foot.

* Compare the average prices by neighborhood.

* Build an interactive neighborhood map.

* Compose your data story.

### Calculate and Plot the Housing Units per Year

For this part of the assignment, use numerical and visual aggregation to calculate the number of housing units per year, and then visualize the results as a bar chart. To do so, complete the following steps:

1. Use the `groupby` function to group the data by year. Aggregate the results by the `mean` of the groups.

2. Use the `hvplot` function to plot the `housing_units_by_year` DataFrame as a bar chart. Make the x-axis represent the `year` and the y-axis represent the `housing_units`.

3. Style and format the line plot to ensure a professionally styled visualization.

4. Note that your resulting plot should appear similar to the following image:

![A screenshot depicts an example of the resulting bar chart.](Images/zoomed-housing-units-by-year.png)

5. Answer the following question:

    * What’s the overall trend in housing units over the period that you’re analyzing?

### Calculate and Plot the Average Sale Prices per Square Foot

For this part of the assignment, use numerical and visual aggregation to calculate the average prices per square foot, and then visualize the results as a bar chart. To do so, complete the following steps:

1. Group the data by year, and then average the results. What’s the lowest gross rent that’s reported for the years that the DataFrame includes?

2. Create a new DataFrame named `prices_square_foot_by_year` by filtering out the “housing_units” column. The new DataFrame should include the averages per year for only the sale price per square foot and the gross rent.

3. Use hvPlot to plot the `prices_square_foot_by_year` DataFrame as a line plot.

    > **Hint** This single plot will include lines for both `sale_price_sqr_foot` and `gross_rent`.

4. Style and format the line plot to ensure a professionally styled visualization.

5. Note that your resulting plot should appear similar to the following image:

![A screenshot depicts an example of the resulting plot.](Images/avg-sale-px-sq-foot-gross-rent.png)

6. Use both the `prices_square_foot_by_year` DataFrame and interactive plots to answer the following questions:

    * Did any year experience a drop in the average sale price per square foot compared to the previous year?

    * If so, did the gross rent increase or decrease during that year?

### Compare the Average Sale Prices by Neighborhood

For this part of the assignment, use interactive visualizations and widgets to explore the average sale price per square foot by neighborhood. To do so, complete the following steps:

1. Create a new DataFrame that groups the original DataFrame by year and neighborhood. Aggregate the results by the `mean` of the groups.

2. Filter out the “housing_units” column to create a DataFrame that includes only the `sale_price_sqr_foot` and `gross_rent` averages per year.

3. Create an interactive line plot with hvPlot that visualizes both `sale_price_sqr_foot` and `gross_rent`. Set the x-axis parameter to the year (`x="year"`). Use the `groupby` parameter to create an interactive widget for `neighborhood`.

4. Style and format the line plot to ensure a professionally styled visualization.

5. Note that your resulting plot should appear similar to the following image:

![A screenshot depicts an example of the resulting plot.](Images/pricing-info-by-neighborhood.png)

6. Use the interactive visualization to answer the following question:

    * For the Anza Vista neighborhood, is the average sale price per square foot for 2016 more or less than the price that’s listed for 2012? 

### Build an Interactive Neighborhood Map

For this part of the assignment, explore the geospatial relationships in the data by using interactive visualizations with hvPlot and GeoViews. To build your map, use the `sfo_data_df` DataFrame (created during the initial import), which includes the neighborhood location data with the average prices. To do all this, complete the following steps:

1. Read the `neighborhood_coordinates.csv` file from the `Resources` folder into the notebook, and create a DataFrame named `neighborhood_locations_df`. Be sure to set the `index_col` of the DataFrame as “Neighborhood”.

2. Using the original `sfo_data_df` Dataframe, create a DataFrame named `all_neighborhood_info_df` that groups the data by neighborhood. Aggregate the results by the `mean` of the group.

3. Review the two code cells that concatenate the `neighborhood_locations_df` DataFrame with the `all_neighborhood_info_df` DataFrame. Note that the first cell uses the [Pandas concat function](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.concat.html) to create a DataFrame named `all_neighborhoods_df`. The second cell cleans the data and sets the “Neighborhood” column. Be sure to run these cells to create the `all_neighborhoods_df` DataFrame, which you’ll need to create the geospatial visualization.

4. Using hvPlot with GeoViews enabled, create a `points` plot for the `all_neighborhoods_df` DataFrame. Be sure to do the following:

    * Set the `geo` parameter to True.
    * Set the `size` parameter to “sale_price_sqr_foot”.
    * Set the `color` parameter to “gross_rent”.
    * Set the `frame_width` parameter to 700.
    * Set the `frame_height` parameter to 500.
    * Include a descriptive title.

Note that your resulting plot should appear similar to the following image:

![A screenshot depicts an example of a scatter plot created with hvPlot and GeoViews.](Images/6-4-geoviews-plot.png)

5. Use the interactive map to answer the following question:

    * Which neighborhood has the highest gross rent, and which has the highest sale price per square foot?

### Compose Your Data Story

Based on the visualizations that you created, answer the following questions:

* How does the trend in rental income growth compare to the trend in sales prices? Does this same trend hold true for all the neighborhoods across San Francisco?

* What insights can you share with your company about the potential one-click, buy-and-rent strategy that they're pursuing? Do neighborhoods exist that you would suggest for investment, and why?

In [36]:
# Import the required libraries and dependencies
import pandas as pd
import hvplot.pandas
from pathlib import Path

## Import the data 

In [37]:
# Using the read_csv function and Path module, create a DataFrame 
# by importing the sfo_neighborhoods_census_data.csv file from the Resources folder
sfo_data_df = pd.read_csv(Path("./Resources/sfo_neighborhoods_census_data.csv"))

# Review the first and last five rows of the DataFrame
display(sfo_data_df.head())
display(sfo_data_df.tail())

year      neighborhood  sale_price_sqr_foot  housing_units  gross_rent
0  2010      Alamo Square           291.182945         372560        1239
1  2010        Anza Vista           267.932583         372560        1239
2  2010           Bayview           170.098665         372560        1239
3  2010  Buena Vista Park           347.394919         372560        1239
4  2010  Central Richmond           319.027623         372560        1239

year            neighborhood  sale_price_sqr_foot  housing_units  \
392  2016          Telegraph Hill           903.049771         384242   
393  2016              Twin Peaks           970.085470         384242   
394  2016  Van Ness/ Civic Center           552.602567         384242   
395  2016       Visitacion Valley           328.319007         384242   
396  2016           Westwood Park           631.195426         384242   

     gross_rent  
392        4390  
393        4390  
394        4390  
395        4390  
396        4390

---

## Calculate and Plot the Housing Units per Year

For this part of the assignment, use numerical and visual aggregation to calculate the number of housing units per year, and then visualize the results as a bar chart. To do so, complete the following steps:

1. Use the `groupby` function to group the data by year. Aggregate the results by the `mean` of the groups.

2. Use the `hvplot` function to plot the `housing_units_by_year` DataFrame as a bar chart. Make the x-axis represent the `year` and the y-axis represent the `housing_units`.

3. Style and format the line plot to ensure a professionally styled visualization.

4. Note that your resulting plot should appear similar to the following image:

![A screenshot depicts an example of the resulting bar chart.](Images/zoomed-housing-units-by-year.png)

5. Answer the following question:

    * What’s the overall trend in housing units over the period that you’re analyzing?



### Step 1: Use the `groupby` function to group the data by year. Aggregate the results by the `mean` of the groups.

In [38]:
# Create a numerical aggregation that groups the data by the year and then averages the results.
housing_units_by_year = round(sfo_data_df.groupby("year").mean(), 2)

# Review the DataFrame
housing_units_by_year

sale_price_sqr_foot  housing_units  gross_rent
year                                                
2010               369.34         372560        1239
2011               341.90         374507        1530
2012               399.39         376454        2324
2013               483.60         378401        2971
2014               556.28         380348        3528
2015               632.54         382295        3739
2016               697.64         384242        4390

In [40]:
min_housing = housing_units_by_year["housing_units"].min()
max_housing = housing_units_by_year["housing_units"].max()

std_housing = housing_units_by_year["housing_units"].std()

### Step 2: Use the `hvplot` function to plot the `housing_units_by_year` DataFrame as a bar chart. Make the x-axis represent the `year` and the y-axis represent the `housing_units`.

### Step 3: Style and format the line plot to ensure a professionally styled visualization.

In [45]:
# Create a visual aggregation explore the housing units by year
housing_units_by_year.hvplot.bar(x= "year",
                                 y= "housing_units",
                                 title= "Housing Units in San Francisco from 2010 to 2016", 
                                 frame_width= 700,
                                 frame_height= 400,
                                 ylim= (min_housing-std_housing, max_housing+std_housing)
                                ).opts(yformatter="%.0f", color='blue')

:Bars   [year]   (housing_units)

### Step 5: Answer the following question:

**Question:** What is the overall trend in housing_units over the period being analyzed?

**Answer:** The trend has pretty much stay the same since 2010. There is little less than 12000 more housing units in a spread of 7 years. Witch we could say is a small increase. It's less then 2000 units per year, witch would be normal progression.

---

## Calculate and Plot the Average Sale Prices per Square Foot

For this part of the assignment, use numerical and visual aggregation to calculate the average prices per square foot, and then visualize the results as a bar chart. To do so, complete the following steps:

1. Group the data by year, and then average the results. What’s the lowest gross rent that’s reported for the years that the DataFrame includes?

2. Create a new DataFrame named `prices_square_foot_by_year` by filtering out the “housing_units” column. The new DataFrame should include the averages per year for only the sale price per square foot and the gross rent.

3. Use hvPlot to plot the `prices_square_foot_by_year` DataFrame as a line plot.

    > **Hint** This single plot will include lines for both `sale_price_sqr_foot` and `gross_rent`.

4. Style and format the line plot to ensure a professionally styled visualization.

5. Note that your resulting plot should appear similar to the following image:

![A screenshot depicts an example of the resulting plot.](Images/avg-sale-px-sq-foot-gross-rent.png)

6. Use both the `prices_square_foot_by_year` DataFrame and interactive plots to answer the following questions:

    * Did any year experience a drop in the average sale price per square foot compared to the previous year?

    * If so, did the gross rent increase or decrease during that year?



### Step 1: Group the data by year, and then average the results.

In [46]:
# Create a numerical aggregation by grouping the data by year and averaging the results
prices_square_foot_by_year = round(sfo_data_df.groupby("year").mean(), 2)

# Review the resulting DataFrame
prices_square_foot_by_year

sale_price_sqr_foot  housing_units  gross_rent
year                                                
2010               369.34         372560        1239
2011               341.90         374507        1530
2012               399.39         376454        2324
2013               483.60         378401        2971
2014               556.28         380348        3528
2015               632.54         382295        3739
2016               697.64         384242        4390

**Question:** What is the lowest gross rent reported for the years included in the DataFrame?

**Answer:** The lowest was in 2010. The lowest gross rent was 1239

### Step 2: Create a new DataFrame named `prices_square_foot_by_year` by filtering out the “housing_units” column. The new DataFrame should include the averages per year for only the sale price per square foot and the gross rent.

In [47]:
# Filter out the housing_units column, creating a new DataFrame 
# Keep only sale_price_sqr_foot and gross_rent averages per year
prices_square_foot_by_year = prices_square_foot_by_year.drop(columns=["housing_units"])

# Review the DataFrame
prices_square_foot_by_year

sale_price_sqr_foot  gross_rent
year                                 
2010               369.34        1239
2011               341.90        1530
2012               399.39        2324
2013               483.60        2971
2014               556.28        3528
2015               632.54        3739
2016               697.64        4390

### Step 3: Use hvPlot to plot the `prices_square_foot_by_year` DataFrame as a line plot.

> **Hint** This single plot will include lines for both `sale_price_sqr_foot` and `gross_rent`

### Step 4: Style and format the line plot to ensure a professionally styled visualization.


In [48]:
# Plot prices_square_foot_by_year. 
# Inclued labels for the x- and y-axes, and a title.
prices_square_foot_by_year.hvplot(title= "Sales Price pre Square Foot & Average Gross Rent in San Francisco -2016-2020",
                                 xlabel= "Year", 
                                 ylabel= "Gross Rent / Sale Price Per Square Foot",
                                 )

:NdOverlay   [Variable]
   :Curve   [year]   (value)

### Step 6: Use both the `prices_square_foot_by_year` DataFrame and interactive plots to answer the following questions:

**Question:** Did any year experience a drop in the average sale price per square foot compared to the previous year?

**Answer:** Yes, there was a slight drop in 2011. 

**Question:** If so, did the gross rent increase or decrease during that year?

**Answer:** The gross rent kept increasing during that year. 

---

## Compare the Average Sale Prices by Neighborhood

For this part of the assignment, use interactive visualizations and widgets to explore the average sale price per square foot by neighborhood. To do so, complete the following steps:

1. Create a new DataFrame that groups the original DataFrame by year and neighborhood. Aggregate the results by the `mean` of the groups.

2. Filter out the “housing_units” column to create a DataFrame that includes only the `sale_price_sqr_foot` and `gross_rent` averages per year.

3. Create an interactive line plot with hvPlot that visualizes both `sale_price_sqr_foot` and `gross_rent`. Set the x-axis parameter to the year (`x="year"`). Use the `groupby` parameter to create an interactive widget for `neighborhood`.

4. Style and format the line plot to ensure a professionally styled visualization.

5. Note that your resulting plot should appear similar to the following image:

![A screenshot depicts an example of the resulting plot.](Images/pricing-info-by-neighborhood.png)

6. Use the interactive visualization to answer the following question:

    * For the Anza Vista neighborhood, is the average sale price per square foot for 2016 more or less than the price that’s listed for 2012? 


### Step 1: Create a new DataFrame that groups the original DataFrame by year and neighborhood. Aggregate the results by the `mean` of the groups.

In [49]:
# Group by year and neighborhood and then create a new dataframe of the mean values
prices_by_year_by_neighborhood = round(sfo_data_df.groupby(["year", "neighborhood"]).mean(), 2)

# Review the DataFrame
prices_by_year_by_neighborhood.head()

sale_price_sqr_foot  housing_units  gross_rent
year neighborhood                                                    
2010 Alamo Square                   291.18         372560        1239
     Anza Vista                     267.93         372560        1239
     Bayview                        170.10         372560        1239
     Buena Vista Park               347.39         372560        1239
     Central Richmond               319.03         372560        1239

### Step 2: Filter out the “housing_units” column to create a DataFrame that includes only the `sale_price_sqr_foot` and `gross_rent` averages per year.

In [50]:
# Filter out the housing_units
prices_by_year_by_neighborhood = prices_by_year_by_neighborhood.drop(columns=["housing_units"])

# Review the first and last five rows of the DataFrame
display(prices_by_year_by_neighborhood.head())
display(prices_by_year_by_neighborhood.tail())

sale_price_sqr_foot  gross_rent
year neighborhood                                     
2010 Alamo Square                   291.18        1239
     Anza Vista                     267.93        1239
     Bayview                        170.10        1239
     Buena Vista Park               347.39        1239
     Central Richmond               319.03        1239

sale_price_sqr_foot  gross_rent
year neighborhood                                           
2016 Telegraph Hill                       903.05        4390
     Twin Peaks                           970.09        4390
     Van Ness/ Civic Center               552.60        4390
     Visitacion Valley                    328.32        4390
     Westwood Park                        631.20        4390

### Step 3: Create an interactive line plot with hvPlot that visualizes both `sale_price_sqr_foot` and `gross_rent`. Set the x-axis parameter to the year (`x="year"`). Use the `groupby` parameter to create an interactive widget for `neighborhood`.

### Step 4: Style and format the line plot to ensure a professionally styled visualization.

In [51]:
# Use hvplot to create an interactive line plot of the average price per square foot
# The plot should have a dropdown selector for the neighborhood
prices_by_year_by_neighborhood.hvplot(groupby= "neighborhood",
                                      x= "year",
                                      ylabel= "Gross Rent / Sale Price Per Square Foot",
                                      title= "Sales Price per Square Foot and Annual Gross Rent -2010-2016 per Neighborhood",
                                      frame_width= 600, 
                                      frame_height= 400)

:DynamicMap   [neighborhood]
   :NdOverlay   [Variable]
      :Curve   [year]   (value)

### Step 6: Use the interactive visualization to answer the following question:

**Question:** For the Anza Vista neighborhood, is the average sale price per square foot for 2016 more or less than the price that’s listed for 2012? 

**Answer:** The price per square foot for the Anza Vista neighborhood in 2016 is 88.4 and it was 344.49 in 2012. 
            So significantly less. 

---

## Build an Interactive Neighborhood Map

For this part of the assignment, explore the geospatial relationships in the data by using interactive visualizations with hvPlot and GeoViews. To build your map, use the `sfo_data_df` DataFrame (created during the initial import), which includes the neighborhood location data with the average prices. To do all this, complete the following steps:

1. Read the `neighborhood_coordinates.csv` file from the `Resources` folder into the notebook, and create a DataFrame named `neighborhood_locations_df`. Be sure to set the `index_col` of the DataFrame as “Neighborhood”.

2. Using the original `sfo_data_df` Dataframe, create a DataFrame named `all_neighborhood_info_df` that groups the data by neighborhood. Aggregate the results by the `mean` of the group.

3. Review the two code cells that concatenate the `neighborhood_locations_df` DataFrame with the `all_neighborhood_info_df` DataFrame. Note that the first cell uses the [Pandas concat function](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.concat.html) to create a DataFrame named `all_neighborhoods_df`. The second cell cleans the data and sets the “Neighborhood” column. Be sure to run these cells to create the `all_neighborhoods_df` DataFrame, which you’ll need to create the geospatial visualization.

4. Using hvPlot with GeoViews enabled, create a `points` plot for the `all_neighborhoods_df` DataFrame. Be sure to do the following:

    * Set the `size` parameter to “sale_price_sqr_foot”.

    * Set the `color` parameter to “gross_rent”.

    * Set the `size_max` parameter to “25”.

    * Set the `zoom` parameter to “11”.

Note that your resulting plot should appear similar to the following image:

![A screenshot depicts an example of a scatter plot created with hvPlot and GeoViews.](Images/6-4-geoviews-plot.png)

5. Use the interactive map to answer the following question:

    * Which neighborhood has the highest gross rent, and which has the highest sale price per square foot?


### Step 1: Read the `neighborhood_coordinates.csv` file from the `Resources` folder into the notebook, and create a DataFrame named `neighborhood_locations_df`. Be sure to set the `index_col` of the DataFrame as “Neighborhood”.

In [52]:
# Load neighborhoods coordinates data
neighborhood_locations_df = pd.read_csv(Path("./Resources/neighborhoods_coordinates.csv"), index_col= "Neighborhood")

# Review the DataFrame
neighborhood_locations_df.head()

Lat         Lon
Neighborhood                          
Alamo Square     37.791012 -122.402100
Anza Vista       37.779598 -122.443451
Bayview          37.734670 -122.401060
Bayview Heights  37.728740 -122.410980
Bernal Heights   37.728630 -122.443050

### Step 2: Using the original `sfo_data_df` Dataframe, create a DataFrame named `all_neighborhood_info_df` that groups the data by neighborhood. Aggregate the results by the `mean` of the group.

In [53]:
# Calculate the mean values for each neighborhood
all_neighborhood_info_df = round(sfo_data_df.groupby("neighborhood").mean(), 2)

# Review the resulting DataFrame
all_neighborhood_info_df.head()

year  sale_price_sqr_foot  housing_units  gross_rent
neighborhood                                                            
Alamo Square     2013.00               366.02       378401.0     2817.29
Anza Vista       2013.33               373.38       379050.0     3031.83
Bayview          2012.00               204.59       376454.0     2318.40
Bayview Heights  2015.00               590.79       382295.0     3739.00
Bernal Heights   2013.50               576.75       379374.5     3080.33

### Step 3: Review the two code cells that concatenate the `neighborhood_locations_df` DataFrame with the `all_neighborhood_info_df` DataFrame. 

Note that the first cell uses the [Pandas concat function](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.concat.html) to create a DataFrame named `all_neighborhoods_df`. 

The second cell cleans the data and sets the “Neighborhood” column. 

Be sure to run these cells to create the `all_neighborhoods_df` DataFrame, which you’ll need to create the geospatial visualization.

In [54]:
# Using the Pandas `concat` function, join the 
# neighborhood_locations_df and the all_neighborhood_info_df DataFrame
# The axis of the concatenation is "columns".
# The concat function will automatially combine columns with
# identical information, while keeping the additional columns.
all_neighborhoods_df = pd.concat(
    [neighborhood_locations_df, all_neighborhood_info_df], 
    axis="columns",
    sort=False
)

# Review the resulting DataFrame
display(all_neighborhoods_df.head())
display(all_neighborhoods_df.tail())


Lat         Lon     year  sale_price_sqr_foot  \
Alamo Square     37.791012 -122.402100  2013.00               366.02   
Anza Vista       37.779598 -122.443451  2013.33               373.38   
Bayview          37.734670 -122.401060  2012.00               204.59   
Bayview Heights  37.728740 -122.410980  2015.00               590.79   
Bernal Heights   37.728630 -122.443050      NaN                  NaN   

                 housing_units  gross_rent  
Alamo Square          378401.0     2817.29  
Anza Vista            379050.0     3031.83  
Bayview               376454.0     2318.40  
Bayview Heights       382295.0     3739.00  
Bernal Heights             NaN         NaN

Lat        Lon    year  sale_price_sqr_foot  \
Yerba Buena      37.79298 -122.39636  2012.5               576.71   
Bernal Heights        NaN        NaN  2013.5               576.75   
Downtown              NaN        NaN  2013.0               391.43   
Ingleside             NaN        NaN  2012.5               367.90   
Outer Richmond        NaN        NaN  2013.0               473.90   

                 housing_units  gross_rent  
Yerba Buena           377427.5     2555.17  
Bernal Heights        379374.5     3080.33  
Downtown              378401.0     2817.29  
Ingleside             377427.5     2509.00  
Outer Richmond        378401.0     2817.29

In [55]:
# Call the dropna function to remove any neighborhoods that do not have data
all_neighborhoods_df = all_neighborhoods_df.reset_index().dropna()

# Rename the "index" column as "Neighborhood" for use in the Visualization
all_neighborhoods_df = all_neighborhoods_df.rename(columns={"index": "Neighborhood"})

# Review the resulting DataFrame
display(all_neighborhoods_df.head())
display(all_neighborhoods_df.tail())

Neighborhood        Lat         Lon     year  sale_price_sqr_foot  \
0      Alamo Square  37.791012 -122.402100  2013.00               366.02   
1        Anza Vista  37.779598 -122.443451  2013.33               373.38   
2           Bayview  37.734670 -122.401060  2012.00               204.59   
3   Bayview Heights  37.728740 -122.410980  2015.00               590.79   
5  Buena Vista Park  37.768160 -122.439330  2012.83               452.68   

   housing_units  gross_rent  
0       378401.0     2817.29  
1       379050.0     3031.83  
2       376454.0     2318.40  
3       382295.0     3739.00  
5       378076.5     2698.83

Neighborhood       Lat         Lon     year  sale_price_sqr_foot  \
68         West Portal  37.74026 -122.463880  2012.25               498.49   
69    Western Addition  37.79298 -122.435790  2012.50               307.56   
70  Westwood Highlands  37.73470 -122.456854  2012.00               533.70   
71       Westwood Park  37.73415 -122.457000  2015.00               687.09   
72         Yerba Buena  37.79298 -122.396360  2012.50               576.71   

    housing_units  gross_rent  
68      376940.75     2515.50  
69      377427.50     2555.17  
70      376454.00     2250.50  
71      382295.00     3959.00  
72      377427.50     2555.17

In [62]:
all_neighborhoods_df.sort_values("gross_rent").tail()

Neighborhood       Lat        Lon    year  sale_price_sqr_foot  \
36     Merced Heights  37.71993 -122.46595  2014.0               788.84   
59     Silver Terrace  37.73467 -122.40106  2014.0               170.29   
67  Visitacion Valley  37.72874 -122.41098  2014.5               301.47   
3     Bayview Heights  37.72874 -122.41098  2015.0               590.79   
71      Westwood Park  37.73415 -122.45700  2015.0               687.09   

    housing_units  gross_rent  
36       380348.0      3414.0  
59       380348.0      3528.0  
67       381321.5      3657.0  
3        382295.0      3739.0  
71       382295.0      3959.0

In [63]:
all_neighborhoods_df.sort_values("sale_price_sqr_foot").head()

Neighborhood       Lat        Lon    year  sale_price_sqr_foot  \
59     Silver Terrace  37.73467 -122.40106  2014.0               170.29   
24      Hunters Point  37.72551 -122.37178  2012.5               170.62   
2             Bayview  37.73467 -122.40106  2012.0               204.59   
47      Outer Mission  37.72280 -122.43869  2013.5               242.37   
67  Visitacion Valley  37.72874 -122.41098  2014.5               301.47   

    housing_units  gross_rent  
59       380348.0     3528.00  
24       377427.5     2489.00  
2        376454.0     2318.40  
47       379374.5     2995.75  
67       381321.5     3657.00

### Step 4: Using hvPlot with GeoViews enabled, create a `points` plot for the `all_neighborhoods_df` DataFrame. Be sure to do the following:

* Set the `geo` parameter to True.
* Set the `size` parameter to “sale_price_sqr_foot”.
* Set the `color` parameter to “gross_rent”.
* Set the `frame_width` parameter to 700.
* Set the `frame_height` parameter to 500.
* Include a descriptive title.

In [56]:
# Create a plot to analyze neighborhood info
all_neighborhoods_df.hvplot.points("Lon",
                                   "Lat",
                                   geo= True, 
                                   tiles= "OSM",
                                   size= "sale_price_sqr_foot", 
                                   color= "gross_rent",
                                   frame_width= 700, 
                                   frame_height= 500,  
                                   title= "San Francisco Sales Price per Square Foot and Annual Gross Rent")

:Overlay
   .Tiles.I  :Tiles   [x,y]
   .Points.I :Points   [Lon,Lat]   (gross_rent,sale_price_sqr_foot)

### Step 5: Use the interactive map to answer the following question:

**Question:** Which neighborhood has the highest gross rent, and which has the highest sale price per square foot?

**Answer:** Union Square District has the highest sale price per square foot and Westwood Park has the highest gross rent

## Compose Your Data Story

Based on the visualizations that you have created, compose a data story that synthesizes your analysis by answering the following questions:

**Question:**  How does the trend in rental income growth compare to the trend in sales prices? Does this same trend hold true for all the neighborhoods across San Francisco?

**Answer:** Rental incomes are growing much faster than the sales prices. Since 2010, average rent has almost quadruple but average sales price have just under doubled. Mostly, as the tendency is true for all of them. but some did not experience the same multiples. 

**Question:** What insights can you share with your company about the potential one-click, buy-and-rent strategy that they're pursuing? Do neighborhoods exist that you would suggest for investment, and why?

**Answer:** It seems that there is a good opportunity in SF because sales prices have not gone up as much as rents. 
Silver Terrace would be a great choice since it has the lowest sales price and one of the highest gross rent. Or Visitacion Valley would be a good choice as well for the same reason